CRM/Mailing/BAO/MailingJob.php

```diff
@@ -1037,4 +987,65 @@ public function writeToDB(
     return $result;
   }
 
+  /**
+   * @param int $jobId
+   * @param string $medium
+   *   Ex: 'email' or 'sms'.
+   * @return \CRM_Mailing_Event_BAO_Queue
+   *   A query object whose rows provide ('id', 'contact_id', 'hash') and ('email' or 'phone').
+   */
+  public static function findPendingTasks($jobId, $medium) {
```

Comment:
Yeah, that's actually the main purpose of the commit. FlexMailer will want to use the exact same query -- which can be accomplished by pulling the query into its own public function.